src/main/java/uk/ac/cam/cl/dtg/segue/api/monitors/AuditMonitor.java

```diff
@@ -0,0 +1,38 @@
+package uk.ac.cam.cl.dtg.segue.api.monitors;
+
+import com.google.inject.Inject;
+import org.jboss.resteasy.spi.HttpRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.ac.cam.cl.dtg.segue.api.managers.UserAuthenticationManager;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.ext.Provider;
+import java.io.IOException;
+
+@Provider
+public class AuditMonitor implements ContainerResponseFilter {
+    private static final Logger log = LoggerFactory.getLogger(AuditMonitor.class);
+
+    private final UserAuthenticationManager userAuthenticationManager;
+
+    @Context private HttpRequest request;
+    @Context private HttpServletRequest httpServletRequest;
+
+    @Inject
+    public AuditMonitor(final UserAuthenticationManager userAuthenticationManager) {
+        this.userAuthenticationManager = userAuthenticationManager;
+    }
+
+    @Override
+    public void filter(ContainerRequestContext containerRequestContext, ContainerResponseContext containerResponseContext) {
+        String plausibleUserId = userAuthenticationManager.getPlausibleUserIdentifierFromCookie(httpServletRequest);
+        log.trace(String.format("User (%s) requested %s \"%s\" with response: %d [%s]",
```

Comment:
Could this not be in some more machine-readable format? One thing that took ages parsing the Nginx logs the other day was corralling them into a format I could open with Excel and use in Python. Comma separated might be enough, since none of these things look to be allowed to have commas in.
I can't imagine we'll ever want to read it by hand, but I do think we should reivew it in an aggregated form regularly.

I was sort of hoping for `ip_address, jsessionid, segue_user_id, http_method, request_path, response_code`.

How will URL parameters interact? Search for instance might get messy if we log the URL param string. We might also want the canonical endpoint URL, like we have for Prometheus, so we can aggregate/search by endpoint more readily without having to worry about different values?

Reply:
OK yes that is sensible and achievable. I've noticed search in the perf logs sometimes being rather long! We should add a word limit to that ES query. Only recording the canonical endpoint would leave out information which might be important. Might be a good idea to move search from using a URL param string to query string.
Revisiting this will let me add documentation to some of these methods too.

Reply:
I actually meant to log both the canonical endpoint and the specific request URL path
`ip_address, jsessionid, segue_user_id, http_method, canonical_path, request_path, response_code`
since it's not _that_ much more additional space.

Reply:
I guess we could also log both the `segue_user_id` from the cookie and whether the HMAC was valid as separate items, in case we wanted to have logs that would let us investigate session-guessing brute force attacks / attempted spoofing. We still can't do session-reuse with an expired session token this way, but that's a much nicher and smaller attack surface.

`ip_address, jsessionid, segue_user_id, session_token, valid_hmac, http_method, canonical_path, request_path, response_code`